plugins/amqp/amqp_input.go

```diff
@@ -203,7 +203,8 @@ func (ai *AMQPInput) Run(ir InputRunner, h PluginHelper) (err error) {
 		}
 		msg.Ack(false)
 	}
-	return nil
+	// return an error message to trigger a potential restart of the plugin
+	return fmt.Errorf("Channel closed while reading from queue %s",  ai.config.Queue)
```

Comment:
So I tested again. Indeed the Stop() method will close the AMQP channel and the Run() method will return an error but the plugin runner detects that Heka is shutting down and it doesn't try to respawn the plugin [0].

[0] https://github.com/mozilla-services/heka/blob/versions/0.10/pipeline/plugin_runners.go#L328